templates/glossary.md

```diff
@@ -0,0 +1,331 @@
+
+### olean file
+
+Lean code has to be compiled.
+The compiled version of file `x.lean` is file `x.olean` and all olean files together form the [cache](#cache).
+
+### orange bar of hell
+
+In VScode, the *orange bar of hell* refers to the orange bar that appears left to the current file when it persists.
+The reason is that the Lean extension has to (re)compile all the imported files whose [cache](#cache) does not match.
+Because of this, having branches that touch two files far apart (with respect to the [import tree](#import-tree)) related through imports is considered bad practice as **any** modification on the file upstream will force Lean to recompile all the files in the middle.
+
+## propeq
+
+*Propositional equality*. Two [term](#term)s `a b : α` are propositionally equal if we can prove `a = b`.
+This is weaker than [definitional](#defeq) and [syntactical](#syntactical-equality) equalities.
+
+#### See also
+
+[Equality, specifications and implementations](https://xenaproject.wordpress.com/2020/07/03/equality-specifications-and-implementations/), from the Xena Project
+
+### `simp` lemma
+
+A lemma which has been tagged with an [attribute](#attribute) which enables it for use with the `simp` tactic.
+
+Good `simp` lemmas guide the `simp` tactic towards reducing complex expressions into simpler ones, often to the point where the `simp` tactic is able to close many goals itself.
+
+Further details can be found in the [`simp` documentation of mathlib](https://leanprover-community.github.io/extras/simp.html#simp-lemmas).
+
+### `simp`-normal form
+
+A convention within [mathlib](#mathlib) for expressing propositions with multiple equivalent forms in a single conventional one.
+
+Examples and further detail can found on [the `simp` page](simp.html#simp-normal-form).
+
+### style linter
+
+A *style linter* is a [linter](#lint) concerned with how code looks.
+Concretely, this is [a short Python program](https://github.com/leanprover-community/mathlib/blob/master/scripts/lint-style.py) checking that all lines are less than `100` characters long, that every file has a [module docstring](#module-docstring)... Errors are flagged in the [style exceptions file](https://github.com/leanprover-community/mathlib/blob/master/scripts/style-exceptions.txt).
+
+### tactic mode
+
+A Lean [mode](#mode) characterized by its reliance on sequences of [tactics](#tactic) which often facilitate proofs quite similar to paper-based reasoning, albeit often with the use of sophisticated tactics which automate tedious portions of a proof.
+There are various means to [enter tactic mode](https://leanprover.github.io/theorem_proving_in_lean/tactics.html#entering-tactic-mode).
+It may be entered using the `by` keyword from [term mode](#term-mode), though in Lean 3 it is most often entered via a `begin...end` block whenever its body is made up of multiple commands.
```

Comment:
```suggestion
In Lean 3 it is most often entered via a `begin...end` block. The `by` keyword can also be used to switch from [term mode](#term-mode) to tactic mode for a single tactic.
```
It feels strange to me to describe the most common mode second.

Reply:
I don't feel strongly, either is fine.

Reply:
Ok, will leave as-is then, thanks!